### ISSUE TITLE
Heroku & Webpack error after build (4.0.0rc)

### ISSUE THREAD

**Issue by arkdelkaos** [angular-fullstack/generator-angular-fullstack#2121]
I'm having the same problem after deploying to my heroku dyno :?

````
$ npm start

> eecr@0.3.0 start /Users/Ark/Documents/Proyectos/EECR/dist
> node server

module.js:442
    throw err;
    ^

Error: Cannot find module '../../webpack.make'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.exports.default (express.js:70:29)
    at Object.<anonymous> (app.js:52:1)
    at Module._compile (module.js:541:32)
    at loader (/Users/Ark/Documents/Proyectos/EECR/dist/node_modules/babel-register/lib/node.js:148:5)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/Ark/Documents/Proyectos/EECR/dist/node_modules/babel-register/lib/node.js:158:7)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/Ark/Documents/Proyectos/EECR/dist/server/index.js:13:28)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.runMain (module.js:575:10)
    at run (bootstrap_node.js:352:7)

npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v6.3.1
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! eecr@0.3.0 start: `node server`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the eecr@0.3.0 start script 'node server'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the eecr package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node server
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs eecr
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls eecr
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/Ark/Documents/Proyectos/EECR/dist/npm-debug.log
````
` /Users/Ark/Documents/Proyectos/EECR/dist/npm-debug.log`
`````
$ cat  /Users/Ark/Documents/Proyectos/EECR/dist/npm-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'start' ]
2 info using npm@3.10.3
3 info using node@v6.3.1
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle eecr@0.3.0~prestart: eecr@0.3.0
6 silly lifecycle eecr@0.3.0~prestart: no script for prestart, continuing
7 info lifecycle eecr@0.3.0~start: eecr@0.3.0
8 verbose lifecycle eecr@0.3.0~start: unsafe-perm in lifecycle true
9 verbose lifecycle eecr@0.3.0~start: PATH: /usr/local/lib/node_modules/npm/bin/node-gyp-bin:/Users/Ark/Documents/Proyectos/EECR/dist/node_modules/.bin:/usr/local/bin:/opt/local/bin:/opt/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin
10 verbose lifecycle eecr@0.3.0~start: CWD: /Users/Ark/Documents/Proyectos/EECR/dist
11 silly lifecycle eecr@0.3.0~start: Args: [ '-c', 'node server' ]
12 silly lifecycle eecr@0.3.0~start: Returned: code: 1  signal: null
13 info lifecycle eecr@0.3.0~start: Failed to exec start script
14 verbose stack Error: eecr@0.3.0 start: `node server`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:242:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:191:7)
14 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:191:7)
14 verbose stack     at maybeClose (internal/child_process.js:852:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
15 verbose pkgid eecr@0.3.0
16 verbose cwd /Users/Ark/Documents/Proyectos/EECR/dist
17 error Darwin 15.6.0
18 error argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
19 error node v6.3.1
20 error npm  v3.10.3
21 error code ELIFECYCLE
22 error eecr@0.3.0 start: `node server`
22 error Exit status 1
23 error Failed at the eecr@0.3.0 start script 'node server'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the eecr package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     node server
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs eecr
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls eecr
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]
````
 - [x] I understand that GitHub issues are not for tech support, but for questions specific to this generator, bug reports, and feature requests.

Item | Version
----- | -----
generator-angular-fullstack | 4.0.0rc
Node | 6.3.1
npm | 3.10.3
Operating System | OS X 10

Item | Answer
----- | -----
Transpiler | Babel
Markup | Jade
CSS | SCSS
Router | ui-router
Client Tests | Mocha
DB | MongoDB
Auth | Yup

**Comment by Awk34** [comment]
Run `node ./server/` instead

**Comment by arkdelkaos** [comment]
`````
$ node ./server/
module.js:442
    throw err;
    ^

Error: Cannot find module '../../webpack.make'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.exports.default (express.js:70:29)
    at Object.<anonymous> (app.js:52:1)
    at Module._compile (module.js:541:32)
    at loader (/Users/Ark/Documents/Proyectos/EECR/dist/node_modules/babel-register/lib/node.js:148:5)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/Ark/Documents/Proyectos/EECR/dist/node_modules/babel-register/lib/node.js:158:7)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/Ark/Documents/Proyectos/EECR/dist/server/index.js:13:28)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.runMain (module.js:575:10)
    at run (bootstrap_node.js:352:7)
````

**Comment by Awk34** [comment]
Oh, I think I know what's happening. `NODE_ENV=production node ./server/`

**Comment by arkdelkaos** [comment]
Premio! :) I didn't tought about that xD I saw that `../../webpack.make` and I panicked xD
But why the init script doesn't send `NODE_ENV=production`by default? It has to, isn't it?
Specially on Heroku...do I have to change `npm start` to `"start": " NODE_ENV=production node server"` manually?

**Comment by Awk34** [comment]
I've never used Heroku, but you just have to make sure that `NODE_ENV` environment variable is set to `production`. I would imagine you could either change the command to include the env var, or there might be a management console where you can set environment variables.

**Comment by arkdelkaos** [comment]
Somehow I thought that the `buildcontrol:heroku`command did it :) I have no problem changing it, of course!
At the management control panel of the heroku admin website I didn't saw where to send it (in fact if you send it as a config variable, it fails)...so I'll try changing the npm start script :)

**Comment by arkdelkaos** [comment]
Well, I have the same problem I had on heroku (that I didn't have on my computer)
````
2016-08-05T19:18:18.321084+00:00 heroku[web.1]: State changed from crashed to starting
2016-08-05T19:18:23.903635+00:00 heroku[web.1]: Starting process with command `npm start`
2016-08-05T19:18:25.990032+00:00 app[web.1]: 
2016-08-05T19:18:25.990057+00:00 app[web.1]: > eecr@0.3.0 start /app
2016-08-05T19:18:26.952317+00:00 heroku[web.1]: State changed from starting to crashed
2016-08-05T19:18:26.943784+00:00 heroku[web.1]: Process exited with status 1
2016-08-05T19:18:26.837554+00:00 app[web.1]: module.js:442
2016-08-05T19:18:26.837565+00:00 app[web.1]:     throw err;
2016-08-05T19:18:26.837566+00:00 app[web.1]:     ^
2016-08-05T19:18:26.837569+00:00 app[web.1]:     at Function.Module._resolveFilename (module.js:440:15)
2016-08-05T19:18:26.837572+00:00 app[web.1]:     at Object.<anonymous> (/app/server/config/express.js:185:19)
2016-08-05T19:18:26.837574+00:00 app[web.1]:     at Object.Module._extensions..js (module.js:550:10)
2016-08-05T19:18:26.837574+00:00 app[web.1]:     at Module.load (module.js:458:32)
2016-08-05T19:18:26.837577+00:00 app[web.1]:     at require (internal/module.js:20:19)
2016-08-05T19:18:26.837578+00:00 app[web.1]:     at Object.<anonymous> (/app/server/app.js:52:1)
2016-08-05T19:18:26.837579+00:00 app[web.1]:     at Module._compile (module.js:541:32)
2016-08-05T19:18:26.837579+00:00 app[web.1]:     at Object.Module._extensions..js (module.js:550:10)
2016-08-05T19:18:26.854199+00:00 app[web.1]: npm ERR! argv "/app/.heroku/node/bin/node" "/app/.heroku/node/bin/npm" "start"
2016-08-05T19:18:26.854326+00:00 app[web.1]: npm ERR! node v6.3.1
2016-08-05T19:18:26.854598+00:00 app[web.1]: npm ERR! eecr@0.3.0 start: `NODE_ENV=production node server`
2016-08-05T19:18:26.854665+00:00 app[web.1]: npm ERR! Exit status 1
2016-08-05T19:18:26.854751+00:00 app[web.1]: npm ERR! 
2016-08-05T19:18:26.854945+00:00 app[web.1]: npm ERR! If you do, this is most likely a problem with the eecr package,
2016-08-05T19:18:26.855024+00:00 app[web.1]: npm ERR! not with npm itself.
2016-08-05T19:18:26.855110+00:00 app[web.1]: npm ERR! Tell the author that this fails on your system:
2016-08-05T19:18:26.855176+00:00 app[web.1]: npm ERR!     NODE_ENV=production node server
2016-08-05T19:18:26.855499+00:00 app[web.1]: npm ERR! There is likely additional logging output above.
2016-08-05T19:18:26.859142+00:00 app[web.1]: 
2016-08-05T19:18:26.859395+00:00 app[web.1]: npm ERR!     /app/npm-debug.log
`````
It didn't help me much with so little information :) But I'll keep looking at it...

**Comment by Awk34** [comment]
`at Object.<anonymous> (/app/server/config/express.js:185:19)`

**Comment by arkdelkaos** [comment]
Yup...but I didn't change a thing of the 'original' express.js file after scaffolding the project :?

**Comment by arkdelkaos** [comment]
It cannot be crashing because I don't have the app registered on npmjs.org...isn't it?  
Do that make any sense? :?
````
0 info it worked if it ends with ok
1 verbose cli [ '/app/.heroku/node/bin/node',
1 verbose cli   '/app/.heroku/node/bin/npm',
1 verbose cli   'bugs',
1 verbose cli   'eecr' ]
2 info using npm@3.10.5
3 info using node@v6.3.1
4 verbose config Skipping project config: /app/.npmrc. (matches userconfig)
5 silly fetchPackageMetaData eecr
6 silly fetchNamedPackageData eecr
7 silly mapToRegistry name eecr
8 silly mapToRegistry using default registry
9 silly mapToRegistry registry https://registry.npmjs.org/
10 silly mapToRegistry data Result {
10 silly mapToRegistry   raw: 'eecr',
10 silly mapToRegistry   scope: null,
10 silly mapToRegistry   escapedName: 'eecr',
10 silly mapToRegistry   name: 'eecr',
10 silly mapToRegistry   rawSpec: '',
10 silly mapToRegistry   spec: 'latest',
10 silly mapToRegistry   type: 'tag' }
11 silly mapToRegistry uri https://registry.npmjs.org/eecr
12 verbose request uri https://registry.npmjs.org/eecr
13 verbose request no auth needed
14 info attempt registry request try #1 at 7:51:55 PM
15 verbose request id d3fa66a88a59728b
16 http request GET https://registry.npmjs.org/eecr
17 http 404 https://registry.npmjs.org/eecr
18 verbose headers { 'content-type': 'application/json',
18 verbose headers   'cache-control': 'max-age=0',
18 verbose headers   'content-length': '2',
18 verbose headers   'accept-ranges': 'bytes',
18 verbose headers   date: 'Fri, 05 Aug 2016 19:51:56 GMT',
18 verbose headers   via: '1.1 varnish',
18 verbose headers   age: '0',
18 verbose headers   connection: 'keep-alive',
18 verbose headers   'x-served-by': 'cache-lcy1125-LCY',
18 verbose headers   'x-cache': 'MISS',
18 verbose headers   'x-cache-hits': '0',
18 verbose headers   'x-timer': 'S1470426716.042966,VS0,VE701',
18 verbose headers   vary: 'Accept-Encoding' }
19 silly get cb [ 404,
19 silly get   { 'content-type': 'application/json',
19 silly get     'cache-control': 'max-age=0',
19 silly get     'content-length': '2',
19 silly get     'accept-ranges': 'bytes',
19 silly get     date: 'Fri, 05 Aug 2016 19:51:56 GMT',
19 silly get     via: '1.1 varnish',
19 silly get     age: '0',
19 silly get     connection: 'keep-alive',
19 silly get     'x-served-by': 'cache-lcy1125-LCY',
19 silly get     'x-cache': 'MISS',
19 silly get     'x-cache-hits': '0',
19 silly get     'x-timer': 'S1470426716.042966,VS0,VE701',
19 silly get     vary: 'Accept-Encoding' } ]
20 silly fetchPackageMetaData Error: Registry returned 404 for GET on https://registry.npmjs.org/eecr
20 silly fetchPackageMetaData     at makeError (/app/.heroku/node/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:264:12)
20 silly fetchPackageMetaData     at CachingRegistryClient.<anonymous> (/app/.heroku/node/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:242:14)
20 silly fetchPackageMetaData     at Request._callback (/app/.heroku/node/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:172:14)
20 silly fetchPackageMetaData     at Request.self.callback (/app/.heroku/node/lib/node_modules/npm/node_modules/request/request.js:200:22)
20 silly fetchPackageMetaData     at emitTwo (events.js:106:13)
20 silly fetchPackageMetaData     at Request.emit (events.js:191:7)
20 silly fetchPackageMetaData     at Request.<anonymous> (/app/.heroku/node/lib/node_modules/npm/node_modules/request/request.js:1067:10)
20 silly fetchPackageMetaData     at emitOne (events.js:101:20)
20 silly fetchPackageMetaData     at Request.emit (events.js:188:7)
20 silly fetchPackageMetaData     at IncomingMessage.<anonymous> (/app/.heroku/node/lib/node_modules/npm/node_modules/request/request.js:988:12)
20 silly fetchPackageMetaData  error for eecr { Error: Registry returned 404 for GET on https://registry.npmjs.org/eecr
20 silly fetchPackageMetaData     at makeError (/app/.heroku/node/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:264:12)
20 silly fetchPackageMetaData     at CachingRegistryClient.<anonymous> (/app/.heroku/node/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:242:14)
20 silly fetchPackageMetaData     at Request._callback (/app/.heroku/node/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:172:14)
20 silly fetchPackageMetaData     at Request.self.callback (/app/.heroku/node/lib/node_modules/npm/node_modules/request/request.js:200:22)
20 silly fetchPackageMetaData     at emitTwo (events.js:106:13)
20 silly fetchPackageMetaData     at Request.emit (events.js:191:7)
20 silly fetchPackageMetaData     at Request.<anonymous> (/app/.heroku/node/lib/node_modules/npm/node_modules/request/request.js:1067:10)
20 silly fetchPackageMetaData     at emitOne (events.js:101:20)
20 silly fetchPackageMetaData     at Request.emit (events.js:188:7)
20 silly fetchPackageMetaData     at IncomingMessage.<anonymous> (/app/.heroku/node/lib/node_modules/npm/node_modules/request/request.js:988:12) pkgid: 'eecr', statusCode: 404, code: 'E404' }
21 verbose stack Error: Registry returned 404 for GET on https://registry.npmjs.org/eecr
21 verbose stack     at makeError (/app/.heroku/node/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:264:12)
21 verbose stack     at CachingRegistryClient.<anonymous> (/app/.heroku/node/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:242:14)
21 verbose stack     at Request._callback (/app/.heroku/node/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:172:14)
21 verbose stack     at Request.self.callback (/app/.heroku/node/lib/node_modules/npm/node_modules/request/request.js:200:22)
21 verbose stack     at emitTwo (events.js:106:13)
21 verbose stack     at Request.emit (events.js:191:7)
21 verbose stack     at Request.<anonymous> (/app/.heroku/node/lib/node_modules/npm/node_modules/request/request.js:1067:10)
21 verbose stack     at emitOne (events.js:101:20)
21 verbose stack     at Request.emit (events.js:188:7)
21 verbose stack     at IncomingMessage.<anonymous> (/app/.heroku/node/lib/node_modules/npm/node_modules/request/request.js:988:12)
22 verbose statusCode 404
23 verbose pkgid eecr
24 verbose cwd /app
25 error Linux 3.13.0-91-generic
26 error argv "/app/.heroku/node/bin/node" "/app/.heroku/node/bin/npm" "bugs" "eecr"
27 error node v6.3.1
28 error npm  v3.10.5
29 error code E404
30 error 404 Registry returned 404 for GET on https://registry.npmjs.org/eecr
31 error 404
32 error 404 'eecr' is not in the npm registry.
33 error 404 You should bug the author to publish it (or use the name yourself!)
34 error 404 Note that you can also install from a
35 error 404 tarball, folder, http url, or git url.
36 verbose exit [ 1, true ]
````

**Comment by peter-holcomb-sp** [comment]
I'm also seeing a similar issue to @arkdelkaos.  Here's what I'm seeing on heroku.  If I comment out the line about strip-ansi and submit another build, it will fail on browser-sync library.  Seems like it just can't find the node modules but i have no idea what's going on:

```
Aug 05 14:30:07 audioplayer-sandbox heroku/web.1:  Starting process with command `npm start` 
Aug 05 14:30:10 audioplayer-sandbox app/web.1:  > audioplayer@0.0.1 start /app 
Aug 05 14:30:10 audioplayer-sandbox app/web.1:  > node server 
Aug 05 14:30:11 audioplayer-sandbox heroku/web.1:  Process exited with status 1 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:  module.js:327 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      throw err; 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      ^ 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:  Error: Cannot find module 'strip-ansi' 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Function.Module._resolveFilename (module.js:325:15) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Function.Module._load (module.js:276:25) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Module.require (module.js:353:17) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at require (internal/module.js:12:17) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Object.<anonymous> (/app/server/config/express.js:174:18) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Module._compile (module.js:409:26) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Object.Module._extensions..js (module.js:416:10) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Module.load (module.js:343:32) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Function.Module._load (module.js:300:12) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Module.require (module.js:353:17) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at require (internal/module.js:12:17) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Object.<anonymous> (/app/server/app.js:36:1) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Module._compile (module.js:409:26) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Object.Module._extensions..js (module.js:416:10) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Module.load (module.js:343:32) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:      at Function.Module._load (module.js:300:12) 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:  npm ERR! Linux 3.13.0-91-generic 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:  npm ERR! argv "/app/.heroku/node/bin/node" "/app/.heroku/node/bin/npm" "start" 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:  npm ERR! node v4.4.7 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:  npm ERR! npm  v2.15.9 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:  npm ERR! code ELIFECYCLE 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:  npm ERR! audioplayer@0.0.1 start: `node server` 
Aug 05 14:30:11 audioplayer-sandbox app/web.1:  npm ERR! Exit status 1 
```

**Comment by arkdelkaos** [comment]
I tried deleting node_modules on server, and re-installing, as I saw at some browser-sync related thread on stackoverflow...but I got the same error messages :(

**Comment by Awk34** [comment]
It's a bug. The browsersync / stripansi imports should be specific to development environment

**Comment by Awk34** [comment]
should be fixed with #2124

**Comment by arkdelkaos** [comment]
Perfect :) Thanks man!
